Merge from upstream

nirarg opened the pull request:
Merge new commits from upstream

ravidbro reviewed (commented):
Is this configMap going to contain specific information about the tenant cluster like clusterID?
If that is the case, the name of the configMap can't be constant since they may be more than one cluster in that namespace.

nirarg commented:
@ravidbro The configMap is located in the tenantcluster and save info about the infracluster (InfraID - is the ID used to create all the resources in the infracluster)

ravidbro commented on pkg/clients/tenantcluster/client.go:
```diff
@@ -36,6 +36,7 @@ const (
 	ConfigMapName             = "cloud-provider-config"
```
Is it in use somewhere else other than the machine-api?
why are we putting it in openshift-config namespace?

nirarg replied:
This information is of the infracluster
We use this configmap, which already created by the installer, because its the place to save the information of the infracluster (like we talked before about how to save the infracluster-namespace)
Currently it not used anywhere else, but I used existing configmap which created by the installer
This is the way the installer can save information which shouldn't be modified by user

ravidbro commented:
/lgtm

chenyosef commented on pkg/clients/tenantcluster/client.go:
```diff
@@ -76,22 +78,42 @@ func (c *kubeClient) GetSecret(secretName string, namespace string) (*corev1.Sec
+	}
+	infraID, ok := (*cMap)[ConfigMapInfraIDKeyName]
+	if !ok {
+		return "", machinecontroller.InvalidMachineConfiguration("Tenant-cluster configMap %s/%s: The map extracted with key %s doesn't contain key %s", ConfigMapNamespace, ConfigMapName, ConfigMapDataKeyName, ConfigMapInfraIDKeyName)
```
What should go wrong for the infraID not to be part of the config map ? Can we maybe indicate the log what cause or how to fix ? The same comment is relevant for other validations of config map keys

nirarg replied:
This is created as part of the platform manifest creation (ignition data) in the installer
I don't think there is case that only this fail and the rest of the modules of the tenantcluster are healthy
If the error happens, the user need to modify the configmap with the right value (but again this not expected, this would happen as part of bigger problem)

chenyosef commented:
/lgtm

ravidbro commented:
/approve

davidvossel committed:
Only inject ssh keys when cloud-config user data is used

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Update machine unit test to require ready condition for detecting is booted

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Removes spec.Bootstrapped in favor of conditions

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Ensure that bootstrap exec condition is set during functional test

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add unit tests for determining if boostrap detection is possible

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add unit test for detecting cloud-config user data type

Signed-off-by: David Vossel <davidvossel@gmail.com>

rmohr committed:
Build and test the code with github actions

Signed-off-by: Roman Mohr <rmohr@redhat.com>

rmohr committed:
Add e2e test lane

Signed-off-by: Roman Mohr <rmohr@redhat.com>

davidvossel committed:
Change IsBooted to IsReady in order to better reflect what is being checked

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #30 from davidvossel/cloud-init-fix

Make ssh bootstrap detection optional

1abhisheksarkar committed:
Update README.md

k8s-ci-robot committed:
Merge pull request #43 from TheAbhishekS/abhishek1

Update README.md

cchengleo committed:
Add dockfile to build manager binary and emit production image

k8s-ci-robot committed:
Merge pull request #38 from rmohr/testing

Add unit- and functional test runners based on github actions

PranshuSrivastava committed:
added meeting info to the README

k8s-ci-robot committed:
Merge pull request #48 from PranshuSrivastava/meeting

added meeting info to the README

rmohr committed:
Make secrets for third-party PRs available

If a PR gets labeled with `ok-to-test` the integration test job is
triggered and this github action has access to secrets.

The github action does for security resaons not allow secrets on all
PRs, so we add this extra step.

Still, the action itself, while using a secret, does not have any
interaction points with users. The PR content is actually not even
checked out.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

k8s-ci-robot committed:
Merge pull request #49 from rmohr/make-secrets-available

Make secrets for third-party PRs available

k8s-ci-robot committed:
Merge pull request #46 from cchengleo/update-dockerfile

Add Dockerfile to build manager binary and emit production image

nunnatsa commented:
/test unit-tests

nunnatsa commented:
/lgtm
/approve
